lib/formats/odata.js

```diff
@@ -527,42 +527,55 @@ const filterFields = (formFields, select, table) => {
 
   let path = '';
 
+  const idKeys = new Set(['__id']);
+
+  // we return parent ID with the subtable, but the key is dynamic
+  // based on the last repeat field
+  let parentIdKey = '__Submissions';
+
   // For subtables we have to include parents fields
   if (table !== 'Submissions') {
-    for (const tableSegment of table.replace(/Submissions\./, '').split('.')) {
+    const tableSegments = table.replace(/Submissions\./, '').split('.');
+    for (let i = 0; i < tableSegments.length; i+=1) {
+      const tableSegment = tableSegments[i];
+
       path += `/${tableSegment}`;
       if (!fieldTree[path]) throw Problem.user.notFound();
       filteredFields.add(fieldTree[path].value);
+
+      if (fieldTree[path].value.type === 'repeat' && i < tableSegments.length-1) {
+        parentIdKey = `__Submissions${path.replace(/\//g, '-')}`;
+      }
     }
+    parentIdKey += '-id';
+    idKeys.add(parentIdKey);
   }
 
   for (const property of select.split(',').map(p => p.trim())) {
     // validate metadata properties. __system/.. properties are only valid for Submission table
-    if (property.startsWith('__id') || property.startsWith('__system')) {
-      if (!(property === '__id' || (table === 'Submissions' && systemFields.has(property))))
-        throw Problem.user.propertyNotFound({ property });
```

Comment:
Could you add a test that `$select`-ing a `__system` property for a subtable results in a 404 response? This `if` used to handle it, but now it's handled by different logic after the `continue`.